localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/integration_request.py

```diff
@@ -26,15 +30,47 @@ def __call__(
 
         if integration_type in (IntegrationType.AWS_PROXY, IntegrationType.HTTP_PROXY):
             # `PROXY` types cannot use integration mapping templates
-            # TODO: check if PROXY can still parameters mapping and substitution in URI for example?
+            # TODO: check if PROXY can still parameters mapping and substitution in URI for example? normally not
             # See
             return
 
-        if integration_type == IntegrationType.MOCK:
-            # TODO: only apply partial rendering of the VTL context
-            return
+        # TODO: maybe first do the `passthroughBehavior` and determine if there's a mapping template?
+
+        # TODO: this handler might not be tested yet until `TemplateMappings` are implemented, as overall behavior will
+        #  change
+        integration_request_parameters = integration["requestParameters"] or {}
+        request_data_mapping = self.get_integration_request_data(
+            context, integration_request_parameters
+        )
+
+        # TODO: work on TemplateMappings with VTL to render the body of the request
+        #  it might populate the context requestOverride too, maybe deepcopy the ContextVariables because it will
+        #  get mutated, or pop it directly? might just be better!
+
+        # TODO: extract the code under into its own method
+
+        # TODO: create helper method to render the different URI with stageVariables and parameters
+        rendered_integration_uri = integration["uri"]  # use request_data_mapping["path"]
+
+        integration_request = IntegrationRequest(
+            http_method=integration["httpMethod"],
```

Comment:
Question: Can this be `ANY`?

Reply:
I'm not sure, and I also see this parameter as not required from the documentation, we'd need to write tests for this: https://docs.aws.amazon.com/apigateway/latest/api/API_PutIntegration.html#API_PutIntegration_RequestSyntax

It might fall back to the default `resourceMethod`? I think we need custom logic here that we need to investigate. I'll update with something temporary with a TODO to come back to this.